azkaban-common/src/test/java/azkaban/executor/RunningExecutionsUpdaterTest.java

```diff
@@ -0,0 +1,158 @@
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class RunningExecutionsUpdaterTest {
+
+  private static final int EXECUTION_ID_77 = 77;
+  private static final ExecutorManagerException API_CALL_EXCEPTION =
+      new ExecutorManagerException("Mocked API timeout");
+
+  @Mock
+  ExecutorManagerUpdaterStage updaterStage;
+  @Mock
+  AlerterHolder alerterHolder;
+  @Mock
+  CommonMetrics commonMetrics;
+  @Mock
+  ExecutorApiGateway apiGateway;
+  @Mock
+  ExecutionFinalizer executionFinalizer;
+  @Mock
+  private Alerter mailAlerter;
+
+  private ExecutableFlow execution;
+  private RunningExecutions runningExecutions;
+  private Executor activeExecutor;
+
+  private RunningExecutionsUpdater updater;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    this.execution = new ExecutableFlow();
+    this.execution.setExecutionId(EXECUTION_ID_77);
+    this.activeExecutor = new Executor(1, "activeExecutor-1", 9999, true);
+    this.runningExecutions = new RunningExecutions();
+    this.runningExecutions.get().put(EXECUTION_ID_77, new Pair<>(
+        new ExecutionReference(EXECUTION_ID_77, this.activeExecutor), this.execution));
+    this.updater = new RunningExecutionsUpdater(this.updaterStage, this.alerterHolder,
+        this.commonMetrics, this.apiGateway, this.runningExecutions, this.executionFinalizer);
+    when(this.alerterHolder.get("email")).thenReturn(this.mailAlerter);
+  }
+
+  @After
+  public void tearDown() {
+    DateTimeUtils.setCurrentMillisSystem();
+  }
+
+  @Test
+  public void updateExecutionsStillRunning() throws Exception {
+    mockFlowStillRunning();
+    this.updater.updateExecutions();
+    verifyCallUpdateApi();
+    verifyZeroInteractions(this.executionFinalizer);
+  }
+
+  @Test
+  public void updateExecutionsExecutorDoesNotExist() throws Exception {
+    mockExecutorDoesNotExist();
+    this.updater.updateExecutions();
+    verifyFinalizeFlow();
+  }
+
+  @Test
+  public void updateExecutionsFlowDoesNotExist() throws Exception {
+    mockFlowDoesNotExist();
+    this.updater.updateExecutions();
+    verifyCallUpdateApi();
+    verifyFinalizeFlow();
+  }
+
+  @Test
+  public void updateExecutionsUpdateCallFails() throws Exception {
+    mockUpdateCallFails();
+    DateTimeUtils.setCurrentMillisFixed(System.currentTimeMillis());
+    for (int i = 0; i < this.updater.numErrorsBeforeUnresponsiveEmail; i++) {
+      this.updater.updateExecutions();
+      DateTimeUtils.setCurrentMillisFixed(
+          DateTimeUtils.currentTimeMillis() + this.updater.errorThreshold + 1L);
+    }
+    verify(this.mailAlerter).alertOnFailedUpdate(
+        this.activeExecutor, Collections.singletonList(this.execution), API_CALL_EXCEPTION);
+
+    // TODO change to checking if executor exist in the DB any more
+    verifyZeroInteractions(this.executionFinalizer);
+    // verify(this.executionFinalizer).finalizeFlow(this.execution, "TODO", null);
```

Comment:
This is to eventually (also) test the improvement where executor is checked again from the DB & it's not found any more. But this test should be also kept for the case where executor exists in the DB but is still not responding.